solutions/task_4.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python3
+"""
+TASK: Implement classes "Temperature", "Person" using Python descriptor protocols to execute unittests without errors.
+"""
+import re
+import unittest
+from datetime import datetime
+
+
+class Temperature(object):
+
+    def __init__(self, value=0):
+        self.fahrenheit = value
+
+    def get_celsius(self):
+        return (self.fahrenheit - 32)/1.8
+
+    def set_celsius(self, value):
+        self.fahrenheit = value*1.8 + 32
+
+    celsius = property(get_celsius, set_celsius)
+
+
+class Descriptor(object):
+
+    def __init__(self, attr_name, attr_type):
+        self.attr_name = attr_name
+        self.attr_type = attr_type
+
+    def __get__(self, instance, owner):
+        return instance.__dict__[self.attr_name]
+
+    def __set__(self, instance, attr_value):
+        if self.attr_type != type(attr_value):
+            raise TypeError
+        if self.attr_name == 'phone':
+            pattern = r"[0-9]{3} [0-9]{2} [0-9]{7}$"
+            if re.match(pattern, attr_value):
+                instance.__dict__.__setitem__(self.attr_name, "+{} ({}) {}-{}-{}".format(attr_value[:3],
+                                                                                         attr_value[4:6],
+                                                                                         attr_value[7:10],
+                                                                                         attr_value[10:12],
+                                                                                         attr_value[12:]))
+            else:
+                raise ValueError
+        else:
+            instance.__dict__.__setitem__(self.attr_name, attr_value)
+
+
+class Person(object):
+    birthday = Descriptor('birthday', datetime)
+    name = Descriptor('name', str)
+    phone = Descriptor('phone', str)
+
+
```

Comment:
Add class:
```
class PersonFieldType(object):
    Birthday = ('birthday', datetime)
    Name = ('name', str)
    Phone = ('phone', str)
```
And thick how to use it and make your code better? (you need it to get rid of "hardcode")

Reply:
May be like this?